lib/tests/test.js

```diff
@@ -144,26 +144,27 @@ class Test {
         self.engine.startService("codeCoverage");
 
         if (self.options.node === 'embark') {
-          return self.engine.ipc.connect((err) => {
-            if (err) {
-              this.engine.logger.error(err.message || err);
-              this.engine.logger.error("Could not connect to Embark's IPC. Is embark running?");
-              process.exit(1);
-            }
-            self.engine.ipc.request('blockchain:node', {}, (err, node) => {
-              if (err) {
-                return self.engine.logger.error(err.message || err);
-              }
-              self.options.node = node;
-              cb();
-            });
-          });
+          if (!self.engine.ipc.connected) {
+            self.engine.logger.error("Could not connect to Embark's IPC. Is embark running?");
+            process.exit(1);
+          }
+          return self.connectToIpcNode(cb);
         }
         cb();
       }
     ], callback);
   }
 
+  connectToIpcNode(cb) {
+    this.engine.ipc.request('blockchain:node', {}, (err, node) => {
+      if (err) {
+        return this.engine.logger.error(err.message || err);
```

Comment:
No callback?

Reply:
I just did an extract function from my code editor here :)
But you're right that seems missing. I will add